packages/content/src/content.ts

```diff
@@ -62,12 +65,121 @@ function getContentData(
   return getItemData(content.id, requestOptions);
 }
 
+interface IMetadataPaths {
+  updateFrequency: string;
+  reviseDate: string;
+  pubDate: string;
+  createDate: string;
+}
+
+export type UpdateFrequency =
+  | "continual"
+  | "daily"
+  | "weekly"
+  | "fortnightly"
+  | "monthly"
+  | "quarterly"
+  | "biannually"
+  | "annually"
+  | "as-needed"
+  | "irregular"
+  | "not-planned"
+  | "unknown"
+  | "semimonthly";
+
+function getMetadataPath(identifier: keyof IMetadataPaths) {
+  // NOTE: i have verified that this will work regardless of the "Metadata Style" set on the org
+  const metadataPaths: IMetadataPaths = {
+    updateFrequency:
+      "metadata.metadata.dataIdInfo.resMaint.maintFreq.MaintFreqCd.@_value",
+    reviseDate: "metadata.metadata.dataIdInfo.idCitation.date.reviseDate",
+    pubDate: "metadata.metadata.dataIdInfo.idCitation.date.pubDate",
+    createDate: "metadata.metadata.dataIdInfo.idCitation.date.createDate"
+  };
+
+  return metadataPaths[identifier];
+}
+
+function getValueFromMetadata(
+  content: IHubContent,
+  identifier: keyof IMetadataPaths
+) {
+  const path = getMetadataPath(identifier);
+  return path && getProp(content, path);
+}
+
+/**
+ * Enriches the content with additional date-related information
+ * Note that this is exported to facilitate testing but it should be considered private
+ *
+ * @private
+ * @param {IHubContent} content - the IHubContent object
+ * @returns {IHubContent}
+ */
+export function _enrichDates(content: IHubContent): IHubContent {
+  const newContent = cloneObject(content);
+
+  // updateFrequency:
+  const updatedFrequencyValue = getValueFromMetadata(
+    newContent,
+    "updateFrequency"
+  );
+  if (updatedFrequencyValue) {
+    const updateFrequencyMap = {
+      "001": "continual",
+      "002": "daily",
+      "003": "weekly",
+      "004": "fortnightly",
+      "005": "monthly",
+      "006": "quarterly",
+      "007": "biannually",
+      "008": "annually",
+      "009": "as-needed",
+      "010": "irregular",
+      "011": "not-planned",
+      "012": "unknown",
+      "013": "semimonthly"
+    } as { [index: string]: UpdateFrequency };
+
+    newContent.updateFrequency = updateFrequencyMap[updatedFrequencyValue];
+  }
+
+  // updatedDate & updatedDateSource:
+  // updatedDate is already set to item.modified, we will override that if we have reviseDate in metadata or lastEditDate
+  const reviseDate = getValueFromMetadata(newContent, "reviseDate");
+  const lastEditDate = getProp(newContent, "layer.editingInfo.lastEditDate");
+  if (reviseDate) {
+    newContent.updatedDate = new Date(reviseDate);
+    newContent.updatedDateSource = getMetadataPath("reviseDate");
+  } else if (lastEditDate) {
+    newContent.updatedDate = new Date(lastEditDate);
+    newContent.updatedDateSource = "layer.editingInfo.lastEditDate";
+  }
```

Comment:
I'm assuming this try-and-fallback is done for reasons.

Reply:
Yep, we want it to use the reviseDate if present and if not fall back to lastEditDate.